### PR TITLE
fix: sanity test cases for CLI installations via Github Actions

### PR DESCRIPTION
**Pull request by shrutiburman**
# Fixes [DII-226](https://issues.corp.twilio.com/browse/DII-226)

This PR: 

- Fixes the testing failures on various installation methods (npm and homebrew) for Twilio CLI. 
- Rather than having a predefined JS test suite, there is an **`sh` file (.github/scripts/cmds-to-test.sh) which has list of commands in it which will be executed** to validate various CLI installations. 
- This shell script can be extended to include any new CLI commands that needs to be verified pertaining to a given release.

Test runs:
[NPM](https://github.com/shrutiburman/twilio-cli/actions/runs/1630507657) installation workflow
[Brew](https://github.com/shrutiburman/twilio-cli/actions/runs/1630528683) installation workflow

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: .github/scripts/cmds-to-test.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+echo "1. twilio --version"
+twilio --version
+echo "2. twilio api:core:addresses:list --help"
+twilio api:core:addresses:list --help
+echo "3. twilio profiles:list"
+twilio profiles:list
+echo "4. twilio help"
+twilio help
+echo "5. twilio api:core:calls --help"
+twilio api:core:calls --help
```
Comment (onuzbee) [inline .github/scripts/cmds-to-test.sh:11]: Nice!

File: .github/workflows/cli-test-homebrew.yml [hunk 2]
```diff
@@ -17,13 +17,17 @@ jobs:
...
+          chmod +x ./.github/scripts/cmds-to-test.sh
+          ./.github/scripts/cmds-to-test.sh
       - name: Global - Testing upgrade Twilio CLI (2)
         run: |
-          brew tap twilio/brew && brew install twilio@${{ github.event.inputs.version }}
+          brew tap twilio/brew && brew install twilio
           brew upgrade twilio
-          node test/asserts/validate_tests.js
+          chmod +x ./.github/scripts/cmds-to-test.sh
+          ./.github/scripts/cmds-to-test.sh
```
Comment (onuzbee) [inline .github/workflows/cli-test-homebrew.yml:32]: [Q] Are we sure here that the action node wouldn't be re-used with any stale caches? 
Asking since if the container is reused internally by Github Actions with a pre-installed CLI, there might be a conflict and we might not be testing the correct version. 
A very edge case- But could we add a sanity step before installation to test `twilio --version` that it errors out? Do you think it's worth the effort?

File: .github/workflows/cli-test-npm.yml [hunk 3]
```diff
@@ -17,22 +17,38 @@ jobs:
...
-         npm make install -g twilio-cli@${{ github.event.inputs.version }}
+         npm install -g twilio-cli@${{ github.event.inputs.version }}
             npm install -g twilio-cli
-            node test/asserts/validate_tests.js
-      - name: Local - Testing installation of Twilio CLI (1)
-        run: |
-         npm install twilio-cli
-         node test/asserts/validate_tests.js
-      - name: Local - Testing upgrade Twilio CLI (2)
+            chmod +x ./.github/scripts/cmds-to-test.sh
```
Comment (onuzbee) [inline .github/workflows/cli-test-npm.yml:38]: Consider using git `--chmod=+x` flag while committing this file to push as an executable to avoid manually changing permissions everytime before execution. Applies everywhere else too

Ref: https://stackoverflow.com/questions/21691202/how-to-create-file-execute-mode-permissions-in-git-on-windows

File: .github/workflows/cli-test-homebrew.yml (hunk 2 above)
Comment (shrutiburman) [reply .github/workflows/cli-test-homebrew.yml:32]: I was worried about the caching too, found out that in order to cache dependencies, actions/cache should be explicitly called out. Also, GH stores caches for ~7 days, so we are covered for the most of it. I did test with this edge case but didn't add it in. 
I'll look at this as an if-else situation so that it errors out with an appropriate error msg.

File: .github/workflows/cli-test-homebrew.yml [hunk 4]
```diff
@@ -17,13 +17,17 @@ jobs:
...
          run: |
-          brew tap twilio/brew && brew install twilio@${{ github.event.inputs.version }}
-          node test/asserts/validate_tests.js
+          brew tap twilio/brew && brew install twilio
+          chmod +x ./.github/scripts/cmds-to-test.sh
+          ./.github/scripts/cmds-to-test.sh
       - name: Global - Testing upgrade Twilio CLI (2)
         run: |
-          brew tap twilio/brew && brew install twilio@${{ github.event.inputs.version }}
+          brew tap twilio/brew && brew install twilio
```
Comment (ravali-rimmalapudi) [inline .github/workflows/cli-test-homebrew.yml:29]: In this case it will install the latest CLI and do the upgrade, but shouldn't we install previous cli and upgrade to latest one?

File: .github/workflows/cli-test-npm.yml [hunk 5]
```diff
@@ -17,22 +17,38 @@ jobs:
...
+        node-version: [16.x, 14.x, 10.x]
+    steps:
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+         node-version: ${{ matrix.node-version }}
+      - name: Local - Testing upgrade Twilio CLI (3)
         run: |
          npm install twilio-cli@${{ github.event.inputs.version }}
-          npm install twilio-cli
```
Comment (ravali-rimmalapudi) [inline .github/workflows/cli-test-npm.yml:36]: Don't we need these commands to test the upgrade?
